modules/rostests/apitests/ntdll/NtSetInformationThread.c

```diff
@@ -0,0 +1,100 @@
+/*
+ * PROJECT:         ReactOS API tests
+ * LICENSE:         GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
+ * PURPOSE:         Tests for the NtSetInformationThread API
+ * COPYRIGHT:       Copyright 2020 George Bișoc <george.bisoc@reactos.org>
+ */
+
+#include "precomp.h"
+
+static
+void
+Test_ThreadPriorityClass(void)
+{
+    NTSTATUS Status;
+    KPRIORITY *Priority;
+
+    Priority = HeapAlloc(GetProcessHeap(), HEAP_ZERO_MEMORY, sizeof(KPRIORITY));
```

Comment:
So, why using a heap allocation instead of stack?

Reply:
Sometimes I just like heap allocation pointers.  😄 
But sure, I'll allocate the variable directly to the stack later.

Reply:
For this test it makes sense, since that way the stack is not corrupted, and the heap can include guard pages around it.